### PR TITLE
feat: Operate OpenAPI explorer (part 2)

### PR DESCRIPTION
**Pull request by pepopowitz**
Supersedes #2867.

## Description

Another attempt at https://github.com/camunda/developer-experience/issues/158.

This PR implements a prototype of embedded-in-the-docs API interactivity for the Operate API. It is based on #2867, and includes multiple changes based on feedback from the first implementation.

```[tasklist]
### Progress
- [x] Duplicate content changes to Operate API Overview from `next` into `8.3`. (e.g. extract Authentication page, remove anything already covered by the interactive explorer.)
- [x] Docs on how to generate the OpenAPI docs
- [x] Make buttons more visible in the API explorer. They are white-bordered, and hard to find.
```

### Post-launch
- [ ] Talk to Operate team about how this affects them moving forward

## Preview site

A preview of the prototype is available [on Netlify](https://loquacious-pika-a05ac2.netlify.app/). [This URL](https://loquacious-pika-a05ac2.netlify.app/api/operate/docs/operate-public-api/) will take you directly to the new Operate API experience.

Note that the preview site uses the docusaurus setting `noIndex: true` to prevent crawlers from indexing it (and potentially introducing duplicate-content issues).

## What it looks like

(But really, the preview site is worth checking out.)

Note that this does not match exactly the preview site. I've added changes to make the buttons stand out a little more, as shown in this screenshot (see caveat below about the buttons).

image

 
## What's been changed in this version

This PR is based on #2867, with several bits of feedback addressed, and some new ideas: 

### Added an "internal link" icon

This is using the existing brand icon that I thought was closest to indicating an internal link. I don't think it's quite right for the job, but maybe others like it. 

image

I suspect that if we want to keep a link here, I will need to requisition an appropriate branded icon.

### Link to API explorer is removed from top nav

Per suggestion, the original top nav is retained. The only remaining way to find the Operate API Explorer is the sidebar: 

image

This will allow us to iterate without confusing users about which APIs are available or considered "first class".

### Operate API overview docs exist in only one place

I've removed the duplication of the Operate API overview page. Some details:

#### The remaining overview exists in the main docs instance, not the API docs instance.

This means [the overview](https://loquacious-pika-a05ac2.netlify.app/docs/next/apis-tools/operate-api/overview/) can be versioned along with the product, which might come in handy if they reference product features only available for specific versions.

I'm totally open to discussion on this. I was very 50/50 on whether the overview should live in the main docs, or directly along-side [the API explorer](https://loquacious-pika-a05ac2.netlify.app/api/operate/docs/operate-public-api/).

The preview site illustrates how this content would remain versioned, as I made more changes to the `next` Operate API Overview docs than the 8.3/8.2 docs (in particular, extracting the Authentication section).

#### There is now a back-link from the API explorer to the main docs Operate API Overview.

Concerned about the separation of the overview content from the API explorer, I added this back-link to hopefully help users find their way back to the main docs: 

image

It links to a specific version, because it _has_ to pick one. I chose to link it to the "current" version.

#### API specs are removed from the overview content (version `next` only)

Details like "endpoints" and "object schemas" are, I think, fully covered by the API explorer. 

More narrative/instructional content about the API specs, like "this is the general pattern we use for Search", remains in the main docs Operate API Overview content.

I only made this change in the `next` version.

#### Sidebar nav reordered for `next` Operate API main docs content

Not a critical piece of this work, but I thought it would be interesting to see if there was a better sequence for the main docs re: Operate API. I only made these changes in the `next` version. This is what the updated content tree looks like: 

image

I'm only going to include this change if it resonates with reviewers.

#### Authentication content extracted from the overview content (version `next` only)

I got excited about https://github.com/camunda/developer-experience/issues/38, and extracted the Auth documentation from the main Overview, but only in the `next` version. This is a step toward consolidating the redundant Authentication content we have throughout our API docs, but it is not a critical piece of this prototype.

### Reduced & reordered the programming languages available in the code sample sidebar

image

The available languages are, in order: 

curl, Java, NodeJS, Go, Python, and C#.

This order is based on [a very timely conversation in slack](https://camunda.slack.com/archives/C0484UFR2VA/p1700554776361249) about popularity of clients.

### Fix the nesting of the object explorer

Things weren't aligned previously. The arrow was on a separate line from the text. Now they're on the same line: 

image

### Caveat: not all tab buttons are bordered

Out of the box, the only bordered tab buttons are the "selected" ones. This is confusing, and it's hard to identify unselected buttons as buttons. 

I added borders to all the non-selected ones, except for one place -- the "language" and "library" selectors in the code sample: 

image

The reason I didn't add borders in this place is multi-fold: 

1. The selectors used to target these elements out of the box are very precise, and hard to override without `!important`.
2. Due to a hack described [in the comments](https://github.com/camunda/camunda-docs/pull/2944#issuecomment-1839176234), there are several caption-less "library" buttons on the page. They are basically invisible without a border, but with a border they become more obvious.
3. I can't swizzle these components to make either of these previous bullet points a valid workaround.

It's not 100% consistent, but I think this is the best option with the tooling we have.

### CODE REVIEW

**Review by akeller — COMMENTED** [review]
I'd love to see more feedback from teams or EMs with APIs, specifically around the topic of versioning. However, I'd also like to move quickly. 

I like that you keep the overview page. I could see if moving to a structure where APIs have an overview page, the API reference (as you are building here), and a single example/user guide that provides more comprehensive examples that what's available in the API reference.

A few additional questions:
* How quickly/easily are we able to turn certain languages on and off? I don't want the existence of a language in our API reference to imply that language is fully supported. I don't _think_ it would (or should) but some people see it and make a lot of leaps.
* Will the future docs on how to generate the OpenAPI docs include how to handle small API changes? Would you handle this in the future or ask teams to update their API reference

**Comment by pepopowitz** [comment]
> I'd love to see more feedback from teams or EMs with APIs, specifically around the topic of versioning. However, I'd also like to move quickly.

I think the best way to prompt feedback might be to launch it and wait for the complaints to file in 😅

**Comment by pepopowitz** [comment]
> * Will the future docs on how to generate the OpenAPI docs include how to handle small API changes? Would you handle this in the future or ask teams to update their API reference

I would expect the docs to explain how to replace the OpenAPI spec, then run a command to regenerate the API's explorer. I wouldn't encourage making small changes in-line, because that feels like it would encourage drift.

Re: who owns the changes -- I asked about this in the previous PR. [Christina mentioned she'd be okay with managing it](https://github.com/camunda/camunda-docs/pull/2867#issuecomment-1801894270), but my preference is for the API owners to update it. I think it doesn't really matter who we initially assign, it will play out like this: 

* Product team notices that the explorer in our docs doesn't match their spec.
* Product team asks us to fix it.
* We nudge them to do it, or establish that we will do it for them.

Regardless, I think I'll plan to set a reminder for myself for a couple weeks before the next release, to nudge each team for updates. Beyond that, I don't feel too strongly about who does the updates.

**Comment by akeller** [comment]
> > I'd love to see more feedback from teams or EMs with APIs, specifically around the topic of versioning. However, I'd also like to move quickly.
> 
> I think the best way to prompt feedback might be to launch it and wait for the complaints to file in 😅

If that's the route you want to go, go for it! 🚀

**Comment by akeller** [comment]
> @akeller How do you feel about the "internal link" icon used in this PR?

No concerns! Doesn't look like our pop-out, external link icon, but looks like it conveys a message that we are going _somewhere_.

**Comment by akeller** [comment]
@pepopowitz how are you doing with this? Are you looking for a final review? It's my last work week of the year, so I won't want you to feel stuck.

**Comment by pepopowitz** [comment]
> @pepopowitz how are you doing with this? Are you looking for a final review? It's my last work week of the year, so I won't want you to feel stuck.

I've got one more thing to push up before final review -- docs on generating the explorer. It's coming soon!

**Comment by pepopowitz** [comment]
@akeller @christinaausley This PR is ready for a final review. My commits from today include things like "docs on generating the explorer", visible borders for buttons, and replicating some content changes in the 8.3 Operate API docs that I'd also made in `next`.

My goal is to merge this week -- I'm also going to reach out to the Operate team in Slack to nudge them to take a look.

**Review by christinaausley — COMMENTED** [review]
I'm going to review the new docs shortly, but one standout is if we capitalize "Explorer" when referencing API Explorer consistently or not. Based on my research, it looks like we should. Do you have any opinions @pepopowitz?

This is an annoying nitpick and shouldn't block this PR, but worth noting so I can add to the WCoE so we don't have to retroactively adjust a year down the road 😅

**Review by christinaausley — DISMISSED** [review]

**Comment by christinaausley** [comment]
Us as a team watching you put this together (nice work Steve :)):

celebration-gif

**Comment by akeller** [comment]
@nloding, you recently made a comment about not wanting to "go look at Swagger". If we offer an experience like this, does your opinion change? No account or Swagger installation is necessary.

**Comment by pepopowitz** [comment]
> I'm going to review the new docs shortly, but one standout is if we capitalize "Explorer" when referencing API Explorer consistently or not. Based on my research, it looks like we should. Do you have any opinions @pepopowitz?
> 
> This is an annoying nitpick and shouldn't block this PR, but worth noting so I can add to the WCoE so we don't have to retroactively adjust a year down the road 😅

I think I changed my mind halfway through this work -- started out capitalizing it, and then eventually decided "it's not a product" and lower-cased it. 

I defer to you! If it should be capitalized, I'll capitalize it everywhere.

File: howtos/documentation-guidelines.md
Comment (akeller) [inline howtos/documentation-guidelines.md:55]: @pepopowitz should there be a table of version associations for the interactive API explorer as well?

File: howtos/interactive-api-explorers.md
Comment (akeller) [inline howtos/interactive-api-explorers.md:26]: Can you add a line about increasing or decreasing the number of languages by contacting ask-developer-experience internally? It may also be good to add something about why those are the chosen languages, but I don't view this as blocking.

**Review by akeller — DISMISSED** [review]
No comment I left is blocking. Feel free to ship and iterate.

**Comment by nloding** [comment]
> @nloding, you recently made a comment about not wanting to "go look at Swagger". If we offer an experience like this, does your opinion change? No account or Swagger installation is necessary.

Yes, I love this experience. Swagger is a useful tool, but it shouldn't be the only way to learn about our APIs. This is a "best of both worlds" type approach IMO. Love it.

File: howtos/documentation-guidelines.md
Comment (pepopowitz) [reply howtos/documentation-guidelines.md:55]: I don't think so! 

The table under "version alignment" shows how main docs versions align with Optimize versions. 

The interactive API explorers will only ever have 1 version, because they're based on the API versions.

File: howtos/interactive-api-explorers.md
Comment (pepopowitz) [reply howtos/interactive-api-explorers.md:26]: I'll do this in a follow-up, unless other changes are also requested (in which case I'll sneak them in with the other changes)!

File: howtos/documentation-guidelines.md
Comment (pepopowitz) [reply howtos/documentation-guidelines.md:55]: Oh wait, I already have other changes from Christina to make! I'll include this change, too.

**Review by christinaausley — APPROVED** [review]
Thank you for those annoying casing adjustments!

**Comment by akeller** [comment]
Thank you for including all of DevEx in the merge party 🎉

**Pull request merged**
